types/vue-apollo.d.ts

```diff
@@ -0,0 +1,77 @@
+import Vue, { PluginObject, PluginFunction } from 'vue';
+import { DocumentNode } from 'graphql';
+import { ApolloClient } from 'apollo-client';
+import { WatchQueryOptions, MutationOptions, SubscriptionOptions, SubscribeToMoreOptions, ObservableQuery, NetworkStatus } from 'apollo-client'
+import { DataProxy } from 'apollo-cache';
+import { subscribe } from 'graphql/subscription/subscribe';
+
+// include Omit type from https://github.com/Microsoft/TypeScript/issues/12215
+type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
+type Omit<T, K extends keyof T> = { [P in Diff<keyof T, K>]?: T[P] };
+
+export class VueApollo implements PluginObject<{}> {
+  [key: string]: any;
+  install: PluginFunction<{}>;
+  constructor (options: {defaultClient: ApolloClient<{}>});
```

Comment:
I guess we also need `defaultOptions` key here.

Reply:
Oh, I missed it. Is this right? 230078a

Reply:
I'm just starting to work with Vue Apollo and I'm not really sure what all `defaultOptions` are. I can't see them in the documentaion. Maybe @Akryum can confirm that.

On the other hand looks like there are more constructor options as mentioned in the docs [here](https://github.com/Akryum/vue-apollo/tree/next#constructor) and please see the source code [here](https://github.com/Akryum/vue-apollo/blob/next/src/apollo-provider.js#L6)

By the way, great work putting this together! 👍

Reply:
https://github.com/Akryum/vue-apollo#special-options